connection.go

```diff
@@ -443,3 +418,25 @@ func GRPCConn(conn Connection) grpc.ClientConnInterface {
 	}
 	return nil
 }
+
+// Helper types for closing lazy clients
+type closeFunc func(ctx context.Context) error
+
+type lazyOnce struct {
```

Comment:
Perfect idea, but I doubt about naming. It is not lazy variant of sync.Once.
It looks like sync about sync inializer/closer.

What about rename ```lazyOnce``` -> ```initOnce``` and ```Do```->```Init```.

code:
```go
	c.ratelimiterOnce.Do(func() closeFunc {
...
		return c.ratelimiter.Close
	})
```
->
```go
	c.ratelimiterOnce.Init(func() closeFunc {
		c.ratelimiter = internalRatelimiter.New(
...
		return c.ratelimiter.Close
	})
```
?